packages/mui-base/src/useTransitionStatus/useTransitionStatus.types.ts

```diff
@@ -0,0 +1 @@
+export type TransitionStatus = 'opening' | 'closing';
```

Comment:
It turns out only two string states are necessary. `undefined` can represent `initial`, where you simply target the base `.Popup` class. This still allows asymmetric transitions:

```js
  if (isRendered && !mounted) {
    setMounted(true);

    if (transitionStatus === 'closing') {
      setTransitionStatus(undefined);
    }
  }
```